Clockies now can summon shields to bash nonbelievers into orbit.

DeltaFire15 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
![example](https://user-images.githubusercontent.com/46569814/81887576-b78c8600-959f-11ea-96e4-12946eb455c0.PNG)

Adds ratvarian shields into the tier two scripture tier.
Gameplaywise, these are simillar to riot shields (except not transparent), though they are repaired with brass instead of titanium, aswell as having high bullet and melee armor in exchange for negative laser and energy resist (as most clockwork gear does),
Though, they do have a more or less unique mechanic, which is that they absorb the damage from blocked attacks (they still damage the shield, so be careful), and allow the user to perform empowered bashes using that energy. Current value for this is a maximum bash multiplier of _x4_ , using _90_ absorbed damage, aka _30_ damage absorbed per multiplier 'tier', leading to about _20_ brute damage, around _60_ stam and _8_ knockback if at the maximum threshhold. 
Bashes will only make use of stored energy if in harm intent, and only actually use it up if at least one poor sod gets hit with the bash. If there is more power than even a bash with highest multiplier would use, it will only use the amount needed to get that multiplier. Having the shield break loses all stored energy.
And no, clockies can't just beat eachother up to generate infinite charge.

Either way, if a nonclockie somehow got their hands on a shield, they can use it to normally bash (charge will never be used though) and block... though there may be a nasty surprise waiting for them if they actually do manage to block an attack.

Shield sprites shamelessly scavenged from clockwork marauders, I ain't the best at spriting so feel free to tell me if you have a better sprite, though this one should suffice.

On another note, this PR also fixes a small thing in the shieldcode itself which only specifically allowed titanium for repairs instead of taking the repair_material var into account.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Clockies as of now are rather lacking of actual combat equipment that is not just utility things (just the spear), leading to them requring to always steal weapons from somewhere, and get absolutely steamrolled if not the case (unless the whole station got converted before the siege phase).
This PR aims at giving clockies a somewhat increased combat capability without needing to ''acquire'' equipment from elsewhere, while introducing a gimmick to the shield that I think would be fun to have (and thematic to the clock cult)
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Clockies can create shields with empowerable bashes.
fix: Shield repairing now works properly for materials other than titanium.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

DeltaFire15 committed:
Shield time

Adds a shield for clockies which empowers bashes via the damage it blocks. Might or might not compile yet. Icons going to be added in the next commit because I'd rather not have to revert all instead of just the icons if I do a fucky

DeltaFire15 committed:
Icon time

Adds the icons for the clockie shield. Just the implementation + icons as of now, no way to get it. Might still not compile :flushed:

DeltaFire15 committed:
Compilation issue resolution

Now it compiles. So it works. Maybe.

DeltaFire15 committed:
Little change

Only harm-intent bashes amplify the bash / use energy.

DeltaFire15 committed:
Fixes up shield.dms repeair proc and moves the brass shield repair there

Oh cool time to throw in a fix aswell, neat!

DeltaFire15 committed:
Some tweaks

Buffs the shields armor and integrity since it was VERY easy to break with energy before, now its medium against those and very strong against ballistics. also a bit easier to accumulate charge because its ''only'' 50% block

DeltaFire15 committed:
Some more stuff for the shield

Adds a spell to summon the shield (20dc time, 600kw cost, tier-two spell)
aswell as adding some ''fun'' for non-clockie users of the shield.

DeltaFire15 committed:
typo man bad

DeltaFire15 committed:
more spellchecking

DeltaFire15 committed:
even more little tweaks

DeltaFire15 committed:
clarification on a comment

DeltaFire15 committed:
sound tweak

only plays a special sound if its an empowered bash

DeltaFire15 committed:
final tweakies before PR

DeltaFire15 committed:
Abuse safeguard

Yeeah might be best to not allow clockies to bash eachother for infinite charge.

DeltaFire15 committed:
another typo fix

I really shouldn't have been coding stuff for this at 5 am...

DeltaFire15 committed:
Safeguards for a VERY specific edge case

Would have caused some weird behavious for edge cases if not resetting the values, so I'm doing it to be sure.

DeltaFire15 committed:
5am code bad

same thing like the previous commit but not unreachable because of return statements.

Linzolle commented on code/modules/antagonists/clockcult/clock_items/clock_weapons/ratvarian_shield.dm:
```diff
@@ -0,0 +1,84 @@
+	var/max_bash_mult = 4
+
+/obj/item/shield/riot/ratvarian/examine(mob/user)
+	if((is_servant_of_ratvar(user) || isobserver(user)))
+		desc = clockwork_desc
+		desc +="\n <span class='inathneq_small'>The shield has absorbed [dam_absorbed] damage, multiplying the effectiveness of its bashes by [calc_bash_mult()]</span>"
+	. = ..()
+	desc = initial(desc)
```
```suggestion
	. = ..()
	if((is_servant_of_ratvar(user) || isobserver(user)))
		. += clockwork_desc
		. +="\n <span class='inathneq_small'>The shield has absorbed [dam_absorbed] damage, multiplying the effectiveness of its bashes by [calc_bash_mult()]</span>"

```

DeltaFire15 replied:
Not feasible as it is supposed to only display clockwork desc if examined by a observer / clockie, not normal desc + clockwork desc.
Clockie / normal desc is also supposed to be above the other special stuff for examine (e.g. materials, if its damaged, etc) ..() does, with only the multiplier acceptable to be below. This means desc needs to be edited, as it is used in ..()

The descriptions in comparison: 
With the change:

![](https://cdn.discordapp.com/attachments/580545840972562454/711650174216437760/unknown.png)

Without the change:

![](https://cdn.discordapp.com/attachments/580545840972562454/711654776311513098/unknown.png)

I can partially comply with this and attach the part about the multiplier via . +=, but desc itself requires the previous implementation. (And the partial change would be quite a bit more ugly as it'd require a check to change desc, then the . = ..() outside the if, and then another if with the same check since it has the same conditions for being displayed)

Ghommie commented on code/modules/antagonists/clockcult/clock_items/clock_weapons/ratvarian_shield.dm:
```suggestion
		. = ..() //If this bash actually hit someone
		if(.)
```

DeltaFire15 replied:
Did this

Ghommie commented on code/modules/antagonists/clockcult/clock_items/clock_weapons/ratvarian_shield.dm:
```suggestion
		return
```

DeltaFire15 replied:
Aaand, logically did that aswell.

Ghommie commented on code/modules/antagonists/clockcult/clock_items/clock_weapons/ratvarian_shield.dm:
I suggest an early return if it is either non-harmful or the user isn't a clockie instead.

DeltaFire15 replied:
Did exactly that.

DeltaFire15 committed:
Applying requested changes / suggestions proposed by Ghommie

Early return statement done by the harmful / servant check, other returns / simillar done via . = ..()

DeltaFire15 committed:
Formatting
